sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/MappedByteBufferInputStream.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * Wraps a {@link ByteBuffer} for access via the {@link InputStream} API.
+ * Supports {@link InputStream#mark(int)} via {@link ByteBuffer#mark()}.
+ */
+public class MappedByteBufferInputStream extends InputStream {
+
+    private final ByteBuffer byteBuffer;
+
+    /**
+     * Creates a new input stream from the given {@link ByteBuffer}.
+     *
+     * @param buffer The buffer to wrap.
+     */
+    public MappedByteBufferInputStream(ByteBuffer buffer) {
+        // Defensive copy of the ByteBuffer
+        this.byteBuffer = ByteBuffer.allocate(buffer.capacity()).put(buffer);
```

Comment:
nit: Use `buffer.remaining()` instead and I recommend using `ByteBuffer.wrap(FluxUtil.byteBufferToArray(buffer))`